core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -2022,7 +2022,7 @@ abstract class RDD[T: ClassTag](
   // TODO: make it public so users can set deterministic level to their custom RDDs.
   // TODO: this can be per-partition. e.g. UnionRDD can have different deterministic level for
   // different partitions.
-  private[spark] final lazy val outputDeterministicLevel: DeterministicLevel.Value = {
+  private[spark] final def outputDeterministicLevel: DeterministicLevel.Value = {
```

Comment:
Should expose this given the TODO?

Reply:
I'd prefer not to expose it here. At least I don't hear a requirement to expose this.

Reply:
Sounds reasonable. Can we add a JIRA to track this and put the JIRA next to the TODO so we eventually come back and make a decision?

Reply:
Yea, sure. Let me add a JIRA here.